Document.cs

```diff
@@ -74,36 +72,68 @@ public class Document : IDisposable
 
 		#region Properties
 
-		DateTime? _ReleaseDate;
-		static readonly object releaseDateLock = new object();
+		private int _majorVersion;
+		private int _minorVersion;
+		private int _pointVersion;
+
 		/// <summary>
-		/// Gets the release date of the underlying Tidy library.
+		/// Returns the major version of libtidy.dll
 		/// </summary>
-		public DateTime ReleaseDate
+		public int MajorVersion
 		{
 			get
 			{
-				lock (releaseDateLock)
-				{
-					if (!_ReleaseDate.HasValue)
-					{
-						DateTime val = DateTime.MinValue;
-						string release = Marshal.PtrToStringAnsi(PInvoke.tidyReleaseDate());
-						if (release != null)
-						{
-							string[] tokens = release.Split(new char[] {' '}, StringSplitOptions.RemoveEmptyEntries);
-							if (tokens.Length >= 3)
-							{
-								DateTime.TryParseExact(tokens[0] + " " + tokens[1] + " " + tokens[2], "d MMMM yyyy", CultureInfo.InvariantCulture, DateTimeStyles.None, out val);
-							}
-						}
-						_ReleaseDate = val;
-					}
-					return _ReleaseDate.Value;
-				}
+				if (_majorVersion != 0) return _majorVersion;
+
+				GetLibraryVersion();
+				return _majorVersion;
+			}
+		}
+
+		/// <summary>
+		/// Returns the minor version of libtidy.dll
+		/// </summary>
+		public int MinorVersion
+		{
+			get
+			{
+				if (_majorVersion != 0) return _minorVersion;
+
+				GetLibraryVersion();
+				return _minorVersion;
 			}
 		}
 
+		/// <summary>
+		/// Returns the point version of libtidy.dll
+		/// </summary>
+		public int PointVersion
+		{
+			get
+			{
+				if (_majorVersion != 0) return _pointVersion;
+
+				GetLibraryVersion();
+				return _pointVersion;
+			}
+		}
+
+		/// <summary>
+		/// Splits the version string into integer parts
+		/// </summary>
+		private void GetLibraryVersion()
+		{
+			var version = Marshal.PtrToStringAnsi(PInvoke.tidyLibraryVersion());
+			if (string.IsNullOrEmpty(version)) return;
+
+			var versionParts = version.Split(new[] {'.'});
+			if (versionParts.Length <= 2) return;
+
+			_majorVersion = int.Parse(versionParts[0]);
+			_minorVersion = int.Parse(versionParts[1]);
+			_pointVersion = int.Parse(versionParts[2]);
```

Comment:
Sure we can't encounter a libtidy version "2.0.5alpha"? Maybe it would be worth a try-catch or using TryParse() since it seems we can live without this if we can't get it.